crates/webidl/src/util.rs

```diff
@@ -707,3 +712,49 @@ pub fn public() -> syn::Visibility {
         pub_token: Default::default(),
     })
 }
+
+/// When generating our web_sys APIs we default to setting slice references that
+/// get passed to JS as mutable in case they get mutated in JS.
+///
+/// In certain cases we know for sure that the slice will not get mutated - for
+/// example when working with the WebGlRenderingContext APIs.
+///
+/// Here we implement a whitelist for those cases. This whitelist is currently
+/// maintained by hand.
+fn maybe_adjust<'a>(mut idl_type: IdlType<'a>, id: &'a OperationId) -> IdlType<'a> {
+    let immutable_f32_fns: Vec<&'static str> = vec![
```

Comment:
This plus the `HashSet` collection below I think can be somewhat costly to do once-per-function, so could this work be executed once and stored into an outer context which is then passed in here?